assets/js/googlesitekit/datastore/site/html.test.js

```diff
@@ -0,0 +1,179 @@
+				const url = 'https://example.com';
+				registry.dispatch( STORE_NAME ).receiveGetHTMLForURL( html, { url } );
+				registry.select( STORE_NAME ).getHTMLForURL( url );
+
+				await subscribeUntil( registry,
+					() => registry.select( STORE_NAME ).hasFinishedResolution( 'getHTMLForURL', [ url ] )
+				);
+
+				registry.dispatch( STORE_NAME ).resetHTMLForURL( url );
+
+				expect( registry.select( STORE_NAME ).hasFinishedResolution( 'getHTMLForURL', [ url ] ) ).toStrictEqual( false );
+			} );
+		} );
+
+		describe( 'receiveGetHTMLForURL', () => {
+			it( 'requires the htmlForURL response', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveGetHTMLForURL();
+				} ).toThrow( 'response is required.' );
+			} );
+
+			it( 'requires the params', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveGetHTMLForURL( '<html>' );
+				} ).toThrow( 'params is required.' );
+			} );
+
+			it( 'receives and sets HTML for a URL ', async () => {
+				const html = '<html><head><title>Example HTML</title></head><body><h1>Example HTML H1</h1></body></html>';
+				const url = 'https://example.com';
+				await registry.dispatch( STORE_NAME ).receiveGetHTMLForURL( html, { url } );
+				expect( registry.select( STORE_NAME ).getHTMLForURL( url ) ).toBe( html );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getHTMLForURL', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				const html = '<html><head><title>Example HTML</title></head><body><h1>Example HTML H1</h1></body></html>';
+				const url = 'https://example.com';
+
+				fetchMock.getOnce(
+					{ query: { tagverify: '1' } },
+					{ body: html, status: 200 }
+				);
+
+				const initialHTML = registry.select( STORE_NAME ).getHTMLForURL( url );
+				// The initialHTML info will be its initial value while the HTML is fetched.
+				expect( initialHTML ).toEqual( undefined );
+				await subscribeUntil( registry, () => registry
+					.select( STORE_NAME )
+					.hasFinishedResolution( 'getHTMLForURL', [ url ] )
+				);
+
+				const selectedHTML = registry.select( STORE_NAME ).getHTMLForURL( url );
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( selectedHTML ).toEqual( html );
+			} );
+
+			it( 'does not make a network request if data is already in state', async () => {
+				const html = '<html><head><title>Example HTML</title></head><body><h1>Example HTML H1</h1></body></html>';
+				const url = 'https://example.com';
+				registry.dispatch( STORE_NAME ).receiveGetHTMLForURL( html, { url } );
+
+				const selectedHTML = registry.select( STORE_NAME ).getHTMLForURL( url );
+
+				await subscribeUntil( registry, () => registry
+					.select( STORE_NAME )
+					.hasFinishedResolution( 'getHTMLForURL', [ url ] )
+				);
+
+				expect( fetchMock ).not.toHaveFetched();
+				expect( selectedHTML ).toEqual( html );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
```

Comment:
Yep, makes sense. I've updated the controller to return the html if it's there (otherwise null), and this test to check that the value is set to null if the request comes back with undefined for the body.